core/memory_tracker/cc/memory_tracker.h

```diff
@@ -311,47 +297,48 @@ class DirtyPageTable {
       current_;  // The space for the last page address stored.
 };
 
-// MemoryTracker utilizes Segfault signal on Linux to track accesses to
+// MemoryTrackerImpl utilizes Segfault signal on Linux to track accesses to
 // memories.
-class MemoryTracker {
+template<typename SpecificMemoryTracker>
```

Comment:
Nit: Not sure if there's any google policies on template variable naming, but I've always used upper-case names, usually single letters, documented in the class.